src/v/pandaproxy/json/requests/fetch.h

```diff
@@ -42,22 +42,44 @@ class rjson_serialize_impl<model::record> {
       , _tpv(tpv)
       , _base_offset(base_offset) {}
 
-    void
+    bool
     operator()(::json::Writer<::json::StringBuffer>& w, model::record record) {
+        auto offset = _base_offset() + record.offset_delta();
+
         w.StartObject();
         w.Key("topic");
         ::json::rjson_serialize(w, _tpv.topic);
         w.Key("key");
-        rjson_serialize_fmt(_fmt)(w, record.release_key());
+        if (!rjson_serialize_fmt(_fmt)(w, record.release_key())) {
+            throw serialize_error(fmt::format(
+              "Unable to serialize key of record at offset {} in "
+              "topic:partition {}:{}",
+              offset,
+              _tpv.topic(),
+              _tpv.partition()));
+        }
         w.Key("value");
-        rjson_serialize_fmt(_fmt)(w, record.release_value());
+        if (!rjson_serialize_fmt(_fmt)(w, record.release_value())) {
+            throw serialize_error(fmt::format(
+              "Unable to serialize value of record at offset {} in "
+              "topic:partition {}:{}",
+              offset,
+              _tpv.topic(),
+              _tpv.partition()));
```

Comment:
Same Q here.
Should we return false instead of throw? IMO, the callsites are where the throw should happen

Reply:
Throwing here allows me to communicate as much data as possible about the serialization error to the user.  If I just return false here, the only place then to report the error is throwing from within `for_each_record` on lines 120-130.

Reply:
That is fair, it is just strange to me that we return bool but only `true` and no `false`

Reply:
yeahhh this is because I had to change the signature for `rjson_serialize_impl<T>::operator()` from `void` to `bool` so I could see that `json_serialize_fmt` was failing when the key or value failed to serialize.